odl/solvers/advanced/proximal_operators.py

```diff
@@ -143,26 +251,28 @@ def make_prox(tau):
 
         return IdentityOperator(space)
 
-    return make_prox
+    return make_identity
 
 
-def proximal_nonnegativity(space):
-    """Function to create the proximal operator of G(x) = ind(x > 0).
+def proximal_clamp(space, lower=None, upper=None):
+    """Function to create the proximal operator of G(x) = ind(a <= x <= b).
 
-    Function for the proximal operator of the functional G(x)=ind(x > 0) to be
-    initialized.
+    Function for the proximal operator of the functional G(x) = ind(a<=x<=b)
+    to be initialized.
 
-    If P is the set of non-negative elements, the indicator function of
+    If P is the set of elements with a < x < b, the indicator function of
     which is defined as
 
-        ind(x > 0) = {0 if x in P, infinity if x is not in P}
+        ind(a < x < b) = {0 if x in P, infinity if x is not in P}
 
     with x being an element in ``space``.
 
-    The proximal operator of G is the point-wise non-negativity thresholding
-    of x
+    The proximal operator of ``tau * G^*`` where ``tau`` is a step size
+    is the point-wise non-negativity thresholding of x
 
-         prox_tau[G](x) = {x if x > 0, 0 if <= 0}
+                              a if x < a,
+         prox[tau * G](x) = { x if a <= x <= b
+                              b if x > b
```

Comment:
This obviously requires the space to be `R^n`. Have you checked the general Hilbert space case? It should be the same, but there could be subtle differences or cases where it's unknown.